internal/services/postgres/postgresql_flexible_server_virtual_endpoint_resource_test.go

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package postgres_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/postgresql/2023-06-01-preview/virtualendpoints"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/postgres"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type PostgresqlFlexibleServerVirtualEndpointResource struct{}
+
+func TestAccPostgresqlFlexibleServerVirtualEndpoint_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_virtual_endpoint", "test")
+	r := PostgresqlFlexibleServerVirtualEndpointResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		data.DisappearsStep(acceptance.DisappearsStepData{
+			Config:       r.basic,
+			TestResource: r,
+		}),
+	})
+}
+
+func TestAccPostgresqlFlexibleServerVirtualEndpoint_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_virtual_endpoint", "test")
+	r := PostgresqlFlexibleServerVirtualEndpointResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.update(data, "azurerm_postgresql_flexible_server.test_replica_0.id"),
```

Comment:
We try to be as explicit as possible in the test configurations and don't pass updated ID values in as arguments to the config. This also fails at the moment:

```
------- Stdout: -------
=== RUN   TestAccPostgresqlFlexibleServerVirtualEndpoint_update
=== PAUSE TestAccPostgresqlFlexibleServerVirtualEndpoint_update
=== CONT  TestAccPostgresqlFlexibleServerVirtualEndpoint_update
    testcase.go:121: Step 1/4 error: Error running pre-apply plan: exit status 1
        Error: length should be equal to or less than 63, got "acctest-psql-virtualendpoint-primary-240807125523377850-replica-0"
          with azurerm_postgresql_flexible_server.test_replica_0,
          on terraform_plugin_test.tf line 51, in resource "azurerm_postgresql_flexible_server" "test_replica_0":
          51:   name                          = "${azurerm_postgresql_flexible_server.test.name}-replica-0"
--- FAIL: TestAccPostgresqlFlexibleServerVirtualEndpoint_update (15.02s)
FAIL
```

Reply:
That's interesting, it was passing locally. Easy enough to shorten the name though.

Would you recommend passing in the full virtual endpoint resources rather than just the ID? Can you point me towards something more in line with what you're looking for?

Reply:
I'm not sure I'll be able to find an example quickly for this specific case, but typically update tests will take advantage of an already existing `basic` or `complete` config, so the acceptance test would like:
```
func TestAccPostgresqlFlexibleServerVirtualEndpoint_update(t *testing.T) {
	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_virtual_endpoint", "test")
	r := PostgresqlFlexibleServerVirtualEndpointResource{}

	data.ResourceTest(t, r, []acceptance.TestStep{
		{
			Config: r.basic(data),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
			),
		},
		data.ImportStep(),
		{
			Config: r.update(data),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
			),
		},
		data.ImportStep(),
	})
}
```

But on second though it's probably sufficient to just shorten the name to prevent the error.

Reply:
I decreased the length of resource names, and the tests all [passed on the TC server](https://hashicorp.teamcity.com/buildConfiguration/TF_AzureRM_AZURERM_SERVICE_PUBLIC_POSTGRES/212499?buildTab=log&focusLine=1000&logView=linear&linesState=973)